core/src/main/java/com/dtolabs/rundeck/core/storage/TypedStorageTreeImpl.java

```diff
@@ -46,8 +46,9 @@ public Resource<ResourceMeta> getResourceWithType(Path path, String contentType)
         if (resource.getContents().getContentType().equals(contentType)) {
             return resource;
         }
+        String errorMessage = "Path " + path + " does not store a password, content-type: " + contentType;
```

Comment:
It's a good fix. 
The original error message is quite confusing; based on my understanding the problem is `Path <path> does not store a resource for content-type: <contentType>`.